src/inc/til/unicode.h

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+namespace til
+{
+    namespace details
+    {
+        inline constexpr wchar_t UNICODE_REPLACEMENT = 0xFFFD;
+    }
+
+    static constexpr bool is_surrogate(const wchar_t wch) noexcept
+    {
+        return (wch & 0xF800) == 0xD800;
+    }
+
+    static constexpr bool is_leading_surrogate(const wchar_t wch) noexcept
+    {
+        return (wch & 0xFC00) == 0xD800;
+    }
+
+    static constexpr bool is_trailing_surrogate(const wchar_t wch) noexcept
+    {
+        return (wch & 0xFC00) == 0xDC00;
+    }
```

Comment:
These constants are a little hard to understand IMO, but they're a direct translation of the bit pattern from the [surrogate pair bit pattern](https://en.wikipedia.org/wiki/UTF-16#Code_points_from_U+010000_to_U+10FFFF). I would've liked to use `0b1111100000000000` instead of `0xF800` but I felt like that would've been a bit too verbose for a mere one-liner.

Reply:
_Yup!_ haha. I used notepad to take a close look at it. This might help for any reviewers that come along:
```
FC00 == 	D800
1111 1100	1101 1000

FC00 == 	DC00
1111 1100	1101 1100


F = 1111
E = 1110
C = 1100
8 = 1000

D800 = |1101 10|00
DC00 = |1101 11|00
DFFF =  1101 11 11

F800 = |1111 1|000
D800 = |1101 1|
```